massgov/openmass
# Create new php script to tag github (major, minor, patch)

**Description:**
Created a new PHP script to tag GitHub (major, minor, patch)

I've added some functionality that might not be needed. The basic function of the GitHub tagging that is needed is there. But I also add major which when its time you will need to add automation to test for major branch using major/ to CircleCI. The release/ and hotfix/ is already there in the CircleCI config I just went with it.

I'm pretty sure this will work in CircleCI, but you might want to test in Openmass to make sure. I also added exit() just in case none of the following words: release, hotfix, or major are found in the commit message it will kill the script with a message.


**Jira:**
https://jira.mass.gov/browse/DP-****


**To Test:**
- [ ] To test this we could add some defined variables: 

Change the release with hotfix or major to see the increment adjust:

```
// This is just a test line to use to check if the following lines work.
$branch = "Merge pull request #206 from massgov/release/0.235.0";
```
Change the 1.0.0 to different tags to see version changes (e.g. 0.235.0)
```
// Adjust the $version test the minor, patch, and major increment changes
$version = new SemVer\Version("1.0.0");
```
Comment out the curl unless you want to test the tagging piece of this PR.

**Screenshots/GIFs:**
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

---

[Peer Review Checklist](https://github.com/massgov/opemass/blob/develop/docs/peer_review_checklist.md)
